File: db_java/src/main/java/com/tut/db/util/Console.java
package com.tut.db.util;

public class Console implements Printable{

    @Override
    public void print(String text) {
        System.out.print(text);
    }

    @Override
    public void fixedPrint(String text, int width, Character fillCharacter) {
        int l = text.length();
        if(l > width)
            System.out.print(text.substring(0, width));
        else{
            int remaining = width - l;
            System.out.print(text);
            for(int i = 0; i < remaining; i++)
                System.out.print(fillCharacter);
        }
    }
}

File: db_kotlin/src/main/kotlin/util/Console.kt
package com.tut.db.util

class Console : Printable {
    override fun print(text: String?) {
        kotlin.io.print(text)
    }

    override fun fixedPrint(text: String?, width: Int, fillChar: Char) {
        val l = text?.length ?: 0;
        if (l > width){
            kotlin.io.print(text?.substring(0, width))
        }
        else{
            val remaining = width - l;
            kotlin.io.print(text)
            for(i in 1..remaining)
                kotlin.io.print(fillChar)
        }
    }

}